lib/router/mbaas.js

```diff
@@ -1,37 +1,68 @@
 'use strict';
 
-var express = require('express')
-  , config = require('../user/config-user')
-  ;
+const express = require('express'),
+  config = require('../user/config-user');
+
 
 function initRouter(mediator) {
+  /**
+   * @param {String} username Unique identifier of the user.
+   * @param {String} password passed in password to be verified against user.
+   * @param {authCallback} cb Node-style callback
+   */
+  var auth = function(username, password, cb) {
+    console.log('Checking credentials for user ' + username);
+    mediator.request('wfm:user:auth', {username: username, password: password}, {uid: username})
+      .then(function(authenticated) {
+        if (authenticated) {
+          console.log(username, ' authenticated');
+          return mediator.request('wfm:user:username:read', username)
```

Comment:
Here you are returning a promise, but below you are calling `cb`.

The function shouldn't do both. It should either use a callback or a promise.

E.g. an example of a series of promises 

```

//Function that returns a promise and resolves it later.
function doSomething() {
   var deferred = q.defer();
   
   //An Async call that resolves the promise after 1 second
   setTimeout(function(){
     deferred.resolve("Finished Successfully");
   }, 1000);
 
   // Immediately returning the promise. Non-async.
   return deferred.promise;
}

function myFunctionReturningAPromise() {
  return mediator.request(...).then(doSomething()).then(doSomethingElse());
}

myFunctionReturningAPromise().then(function(resultOfChain){
  console.log("The promise chain completed fully", resultOfChain);
}).reject(function(error){
   console.log("An Error Occurred In The Chain", error);
});

```

`doSomething` and `doSomethingElse` also return Promises.